test/framework/workload.go

```diff
@@ -54,15 +61,96 @@ func (w *WorkloadCluster) DeleteClusterWithKubectl() {
 	w.StopIfFailed()
 }
 
-func (w *WorkloadCluster) applyClusterManifest(ctx context.Context) {
-	if err := w.KubectlClient.ApplyManifest(ctx, w.ManagementClusterKubeconfigFile(), w.ClusterConfigLocation); err != nil {
-		w.T.Fatalf("Failed to apply workload cluster config: %s", err)
+// CreateClusterWithAPIGitOps uses gitops to create a cluster with the API.
+func (w *WorkloadCluster) CreateClusterWithAPIGitOps() {
+	ctx := context.Background()
+	w.createWithAPIGitOps(ctx)
+	w.waitForKubeconfig(ctx)
+	w.ValidateClusterState()
+	w.T.Logf("Successfully created version controlled cluster configuration")
+}
+
+func (w *WorkloadCluster) createWithAPIGitOps(ctx context.Context) {
+	w.ManagementCluster.initGit(ctx)
```

Comment:
this code is all very entangled
Plus it's not thread safe, you can be possibly modifying the management cluster test concurrently from the workload clusters

This goes back to my comment about giving the workload cluster test access to the management cluster test. I believe we need to rethink this design and decouple them.